src/bentoml/_internal/service/openapi/utils.py

```diff
@@ -43,6 +74,33 @@ def pydantic_components_schema(pydantic_model: t.Type[pydantic.BaseModel]):
     return {k: Schema(**definitions[k]) for k in sorted(definitions)}
 
 
+def typeddict_components_schema(typeddict: t.Type[TypedDict]):
+    typeddict_set = get_flat_typeddicts_from_typeddict(
+        typeddict=typeddict, typeddict_set=set()
+    )
+    definitions: dict[str, Schema] = {}
+    for typeddict in typeddict_set:
+        definitions[typeddict.__name__] = typed_dict_to_dict(typeddict)
+    return {k: Schema(**definitions[k]) for k in sorted(definitions)}
+
+
+def get_flat_typeddicts_from_typeddict(
+    typeddict: t.Type[TypedDict], typeddict_set: t.Set[t.Type[t.Any]]
+) -> t.Set[t.Type[t.Any]]:
+    typeddict_set.add(typeddict)
+    field_types: t.Dict[str, t.Any] = get_type_hints(typeddict, include_extras=True)
```

Comment:
get_type_hints is very expensive, but I guess we don't have any other choice really.

Reply:
IMO `openapi_components()` is only used for `/docs.json` 
it means that there is no issue to inference performance? so I use it 😭

Reply:
The swagger UI depends on this right? so `openapi_components` path would have some penalty for actual loading the UI.

We probably just need to add documentation somewhere about this.